lib/opte/src/engine/port.rs

```diff
@@ -1961,30 +1926,33 @@ impl<N: NetworkImpl> Port<N> {
                             // all other layers' stats.
                             return Ok(ProcessResult::Modified);
                         }
-                        Err(ProcessError::TcpFlow(
-                            e @ TcpFlowStateError::UnexpectedSegment { .. },
-                        )) => {
-                            // Technically unreachable, as we filter these out in `update_tcp_entry`.
-                            // Panicking here would probably be overly fragile, however.
+                        Err(ProcessError::MissingFlow(flow_id)) => {
+                            let e = format!("Missing TCP flow ID: {flow_id}");
                             self.tcp_err(
                                 &data.tcp_flows,
                                 Direction::In,
-                                e.to_string(),
+                                e,
                                 pkt,
                             );
-                            return Ok(ProcessResult::Drop {
-                                reason: DropReason::TcpErr,
-                            });
+                            // If we have a UFT but no TCP flow ID, there is likely a bug
+                            // and we are now out of sync. As above we can't reprocess,
+                            // but we have regenerated the TCP entry to be less disruptive
+                            // than a drop. Remove the UFT entry on the same proviso since the
+                            // next packet to use it will regenerate it.
+                            data.uft_in.remove(ufid_in);
```

Comment:
Use `uft_invalidate` here so we also get the probe?

Reply:
Also, can't comment on it directly (grrr github context...), but mind using `ufid_in` instead of `pkt.flow()` on the match above for consistency.